gobblin-service/src/main/java/org/apache/gobblin/service/modules/spec/JobExecutionPlan.java

```diff
@@ -59,8 +59,7 @@
  * where the {@link JobSpec} will be executed.
  */
 @Data
-@EqualsAndHashCode(exclude = {"executionStatus", "currentAttempts", "jobFuture", "flowStartTime"})
-// todo - consider excluding SpecExecutor from EqualsAndHashCode or only including DagNodeId
+@EqualsAndHashCode(of = {"id"})
```

Comment:
double checking this is the only value needed to identify `JobExecutionPlan` in a set? so if any other prop changes we want to be able to retrieve and delete the `JobExecutionPlan`

Reply:
i believe this should be the only identifier of a dag node.
id is unique, it comprises of fg, fn, eid, jg, jn. cc. @Will-Lo 
one can argue that if two dag nodes having the same id have different properties, they should be considered the same, but then they should not have same id.